fe2-android/app/src/androidTest/java/com/github/dedis/popstellar/ui/detail/event/rollcall/RollCallEventCreationFragmentTest.java

```diff
@@ -1,116 +1,231 @@
-  //
-  //    onView(withId(R.id.start_time_edit_text)).perform(click());
-  //    onView(withClassName(Matchers.equalTo(TimePicker.class.getName())))
-  //        .perform(PickerActions.setTime(HOURS, MINUTES));
-  //    onView(withId(android.R.id.button1)).perform(click());
-  //    onView(withId(R.id.start_time_edit_text)).check(matches(withText(TIME)));
-  //
-  //    onView(withId(R.id.roll_call_confirm)).check(matches(isEnabled()));
-  //    onView(withId(R.id.roll_call_confirm)).perform(click());
-  //
-  //    activityScenarioRule
-  //        .getScenario()
-  //        .onActivity(
-  //            activity -> {
-  //              PoPApplication app = (PoPApplication) activity.getApplication();
-  //              List<Event> events =
-  //                  app.getCurrentLao().map(Lao::getEvents).orElse(new ArrayList<>());
-  //              List<String> eventsName =
-  //                  events.stream().map(Event::getName).collect(Collectors.toList());
-  //              Assert.assertThat(RANDOM_EVENT_TITLE, isIn(eventsName));
-  //            });
-  //  }
+
+  private static final int YEAR;
+  private static final int MONTH_OF_YEAR;
+  private static final int DAY_OF_MONTH;
+  private static final String DATE;
+
+  private static final int HOURS = 12;
+  private static final int MINUTES = 15;
+  private static final String TIME = "" + HOURS + ":" + MINUTES;
+
+  static {
+    // Make sure the date is always in the future
+    Calendar today = Calendar.getInstance();
+    today.add(Calendar.MONTH, 13);
+
+    YEAR = today.get(Calendar.YEAR);
+    MONTH_OF_YEAR = today.get(Calendar.MONTH);
+    DAY_OF_MONTH = today.get(Calendar.DAY_OF_MONTH);
+
+    DATE = DAY_OF_MONTH + "/" + MONTH_OF_YEAR + "/" + YEAR;
+  }
+
+  @Rule
+  public final FragmentScenarioRule<RollCallEventCreationFragment> fragmentRule =
+      FragmentScenarioRule.launchInContainer(RollCallEventCreationFragment.class);
+
+  @Test
+  public void canLaunchDatePickerFragmentFromStartDateButton() {
+    startDateView().perform(click());
+    datePicker().check(matches(isDisplayed()));
+  }
+
+  @Test
+  public void canLaunchDatePickerFragmentFromEndDateButton() {
+    endDateView().perform(click());
+    datePicker().check(matches(isDisplayed()));
+  }
+
+  @Test
+  public void canChooseRandomDate() {
+    startDateView().perform(click());
+
+    datePicker().perform(PickerActions.setDate(YEAR, MONTH_OF_YEAR, DAY_OF_MONTH));
+    pickerAcceptButton().perform(click());
+
+    startDateView().check(matches(withText(DATE)));
+  }
+
+  @Test
+  public void datePickerChoosesTodayByDefault() {
+    final Calendar currentCalendar = Calendar.getInstance();
+    int year = currentCalendar.get(Calendar.YEAR);
+    int month = currentCalendar.get(Calendar.MONTH) + 1;
+    int day = currentCalendar.get(Calendar.DAY_OF_MONTH);
+    final String DATE =
+        (day < 10 ? "0" : "") + day + "/" + (month < 10 ? "0" : "") + month + "/" + year;
```

Comment:
Using the same variable name as the other static variable can be confusing.

Reply:
done